firestore/integration_test_internal/src/document_change_test.cc

```diff
@@ -78,5 +78,67 @@ TEST_F(DocumentChangeTest, TestDocumentChanges) {
 
 #endif  // defined(__ANDROID__)
 
+TEST_F(DocumentChangeTest, Equality) {
+  DocumentChange invalid_change_1 = DocumentChange();
+  DocumentChange invalid_change_2 = DocumentChange();
+
+  EXPECT_TRUE(invalid_change_1 == invalid_change_2);
+  EXPECT_FALSE(invalid_change_1 != invalid_change_2);
+
+  CollectionReference collection = Collection();
+  Query query = collection.OrderBy("a");
+
+  DocumentReference doc1 = collection.Document("1");
+  DocumentReference doc2 = collection.Document("2");
+
+  TestEventListener<QuerySnapshot> listener("TestDocumentChanges");
+  ListenerRegistration registration = listener.AttachTo(&query);
+  Await(listener);
+  QuerySnapshot snapshot = listener.last_result();
+  EXPECT_EQ(snapshot.size(), 0);
+
+  WriteDocument(doc1, MapFieldValue{{"a", FieldValue::Integer(1)}});
+  Await(listener);
+  snapshot = listener.last_result();
+
+  std::vector<DocumentChange> changes = snapshot.DocumentChanges();
+  EXPECT_EQ(changes.size(), 1);
+  // first_change: Added doc1.
+  auto first_change = changes[0];
```

Comment:
Optional: Consider using the names `change1` and `change2` instead of `first_change` and `second_change`, respectively. IMO it's a bit more readable that way and consistent with other naming conventions like `doc1` and `doc2`.

Reply:
+1.